root/static/scripts/edit/components/forms.js

```diff
@@ -59,3 +59,36 @@ MB.initializeArtistCredit = function (form, initialArtistCredit) {
     );
   });
 };
+
+/*
+ * Registers a beforeunload event listener on the window that prompts
+ * the user if any of the page's form inputs have been changed.
+ */
+MB.installFormUnloadWarning = function () {
+  let modified = false;
+
+  const form = document.querySelector('#page form');
+
+  /*
+   * This is somewhat heavy-handed, in that it will still warn even if the
+   * user changes an input back to its original value.
+   */
+  form.addEventListener('change', () => {
+    modified = true;
+  });
+
+  // Disarm the warning when the form is being submitted.
+  form.addEventListener('submit', () => {
+    modified = false;
+  });
+
+  window.addEventListener('beforeunload', event => {
+    if (!modified) {
+      return false;
+    }
+    event.returnValue = l(
```

Comment:
Hmm, I'm not sure if this prompt will cause issues with the Selenium tests...

Reply:
It does, but I think I've fixed all the tests that entered data and then either navigated somewhere else or left the page open for the next page to navigate somewhere else.